airflow/migrations/versions/c306b5b5ae4a_switch_xcom_table_to_use_run_id.py

```diff
@@ -41,7 +41,7 @@
 
 def _get_new_xcom_columns() -> Sequence[Column]:
     return [
-        Column("dagrun_id", Integer(), nullable=False),
+        Column("dag_run_id", Integer(), nullable=False),
```

Comment:
IMO `dagrun_id` is more appropriate here because it refers to the `id` column on the `dagrun` table.

Reply:
the table is actually `dag_run` -- not `dagrun`

Reply:
does that change your feeling about this?

Reply:
I think I'm with TP on this. Yes the table is "dag_run", but by keeping this as "dagrun_id" it is (marginally) clearer that this is the "id" column of the "dagrun" table.

Reply:
i understand, but i disagree.

in my experience, if you have an order_line table, you call the identity column either `order_line_id` or simply `id`, but not `orderline_id`.  you don't violate your snake case convention for identity columns.

and there's good reason we use snake case.  consider table `ab_permission_view_role`.  would you really want to call that table's identity column `abpermissionviewrole_id`?  

actually, looking at that table, its ident col is `id` -- maybe we should just go with that here.

Reply:
Ah i see we're talking about the column in the xcom table, which is pointing to the dag run table.

The PK in dag_run is already `id`.

So yeah, same reasoning essentially applies.  When we refer to id col of a table, it should be f"{table_name}_{col_name}".  I don't think there's really any reason to diverge from this convention here.  There's no "run_id" on the "dag" table so there's no conflict here.

Reply:
> And dag_run is also an outlier in Airflow since most other tables e.g. taskinstance do lack the undercores).

it's not actually true, they are all snake case.  the table for `TaskInstance` is `task_instance`.  and there's not a single table in airflow db that's not snake case table name.  (But the _python module_ names are like that.)

> I guess this is essentially down to personally preference

To some extent that's true but consistency and convention also count for something.  Particularly in the context of database object naming conventions, in this area it's a lot easier to make conventions about than some other decisions we have to make.  And referring to the PK as `f"{table_name}_id` is i think pretty standard[1].  E.g. you would not create a table like this:

```sql
create table dag_run (
dagrun_id int identity,
val varchar
)
```

That would not make any sense.  Why not call the table `dagrun` in that case?  I think it's essentially the same here.  The table is called `dag_run` so it's ID should be referred to as `dag_run_id`.  Since the table names have already made the decision to go snake case, why should the columns go backwards on that decision?  And it's not that we can't ever deviate from the standard choice, but that side of the argument bears the burden; there should be a good reason to make an exception.

As always, I'll go with the majority view,  but just trying to present what makes sense to me.  And of course it's always better if we can reach consensus.  

---

[1] Jetbrains can even [suggest joins](https://www.jetbrains.com/help/datagrip/foreign-keys.html#configure-rules-for-virtual-foreign-keys) for this table naming pattern when no FK is defined.